### PR TITLE
[WIP] Mock Source creating

### PR DESCRIPTION
**Pull request by slemrmartin**

### CODE REVIEW

File: lib/topological_inventory/mock_source/operations/worker.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        attr_accessor :payload
+
+        def create_mock_source
+          source = post_source
+          post_endpoint(source)
+        end
+
+        def post_source
+          source = TopologicalInventoryApiClient::Source.new(
+            :name           => "Mock Source",
```
Comment (agrare) [inline lib/topological_inventory/mock_source/operations/worker.rb:29]: Probably will want the name to be input from the user

File: lib/topological_inventory/mock_source/operations/worker.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            :source_type_id => '1',
```
Comment (agrare) [inline lib/topological_inventory/mock_source/operations/worker.rb:30]: You'll need to look up the source type, not depend on the id

File: lib/topological_inventory/mock_source/operations/worker.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            :tenant_id      => '1',
```
Comment (agrare) [inline lib/topological_inventory/mock_source/operations/worker.rb:31]: This has to be set with the x-rh-identity headers, again not depending on the id

File: lib/topological_inventory/mock_source/operations/worker.rb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            :uid            => SecureRandom.uuid,
```
Comment (agrare) [inline lib/topological_inventory/mock_source/operations/worker.rb:32]: The backend will autogen a uid for you

**Review by agrare — COMMENTED** [review]
@slemrmartin I'm confused, so the operation that this is performing is creating a mock source?  Does that gain us much above someone posting to the API?

Also this looks like a single run script not a worker, maybe a tool/script that assists in posting to the API?

**Comment by slemrmartin** [comment]
@agrare this is only a try, I added WIP. For Mass creating, I'll create probably simple rails app as a new container app.

**Comment by agrare** [comment]
Hm okay I think i'm missing what you're trying to do then.  For mass creation a tool that you could use in a bash script would also help.  Unless you're creating a new UI or API how would a rails app help?

**Comment by slemrmartin** [comment]
@agrare this is parallel discussion to https://github.com/ManageIQ/topological_inventory-core/pull/130 :)

Can I easily access all CI/stage APIs from my local machine by bash script? Can I use one-time running script in container? 
If I create new UI running in container, then it can be used easily by anyone. I don't plan some patternfly graphic interface, just simple few inputs.

**Comment by carbonin** [comment]
> For mass creation a tool that you could use in a bash script would also help

This was my thought as well. Create a tool that uses the API to create the sources. I'm not sure why we would deploy this as a part of the mock source image.

**Comment by carbonin** [comment]
> Can I easily access all CI/stage APIs from my local machine by bash script?

Yes, they're publicly available. And this would also solve the tenant problem if the user of the script provided credentials.

> Can I use one-time running script in container?

I'm not sure why you would want to do this.

 > If I create new UI running in container, then it can be used easily by anyone.

A ruby script is even simpler and then you won't have to host a new UI somewhere.

**Comment by slemrmartin** [comment]
@agrare another "Why UI" points are: 
* I don't need to explain to everyone how it works, how to fill it
* UI container would run all the time
* inputs, especially settings in TextArea can be pre-filled by ConfigMap. 
* check results will need another bash script. it can be in UI container too instead and everyone will understand it without documentation

**Comment by slemrmartin** [comment]
@carbonin
> > Can I easily access all CI/stage APIs from my local machine by bash script?
> 
> Yes, they're publicly available. And this would also solve the tenant problem if the user of the script provided credentials.

Ok, I heard they don't. How? (and how to provide credentials through script? Btw I still don't have access to CI, because I'm waiting for account at least a month). 
> 
> > Can I use one-time running script in container?

It was question when APIs are not publicly available, nevermind. 
> 
> I'm not sure why you would want to do this.
> 
> > If I create new UI running in container, then it can be used easily by anyone.
> 
> A ruby script is even simpler and then you won't have to host a new UI somewhere.

yes, but UI is self-explanatory and ConfigMap-able. I can provide credentials through it in the same way like through script, can't I? I can definitely start with scripts, because potential UI will use almost the same code...but I thought that S2I deployment from github to openshift is not too difficult, just one build config

**Comment by agrare** [comment]
@slemrmartin I think this can be closed now?

**Pull request closed without merging**